### PR TITLE
Addition of feature flags tab in admin page

### PR DESCRIPTION
**Pull request by Courtneyp123**
Admin page feature flag tab addition.  Style included with lint tests and code coverage met.

### CODE REVIEW

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <a  (click)="deleteFeatureFlag(item?.id)">
+          <app-dash-trash class="clickable pull-right"></app-dash-trash>
+        </a>
+        <app-dash-edit class="clickable pull-right" (click)="editFeatureFlag(item)"></app-dash-edit>
+      </div>
+      <h6>Name: {{item?.name}}</h6>
+      <h6>Description: {{item?.description}}</h6>
+      <div class="flags">
+        <h6>Flags: </h6>
+        <div class="table-row" *ngFor="let k of flagKeys(item?.flags)">
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.html:17]: Rename `k` to something more meaningful.

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  addFile() {
+    const modalRef = this.modalService.open(UpdateJsonComponent);
+    modalRef.result.then((newConfig) => {
+      if (newConfig === 'close') {
+        this.loadFeatureFlags();
+      }
+    }).catch((error) => {
+      console.log('add error newConfig: ' + error);
+    });
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts:60]: This block is copied and pasted multiple times.

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/create-or-update-feature-flags/create-or-update-feature-flags.component.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        this.featureFlagForm.get('scm').setValue(true);
+      } else { this.featureFlagForm.get('scm').setValue(false); }
+
+      if (this.flags.staticSecurityScan) {
+        this.featureFlagForm.get('staticSecurityScan').setValue(true);
+      } else { this.featureFlagForm.get('staticSecurityScan').setValue(false); }
+
+      if (this.flags.test) {
+        this.featureFlagForm.get('test').setValue(true);
+      } else { this.featureFlagForm.get('test').setValue(false); }
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/create-or-update-feature-flags/create-or-update-feature-flags.component.ts:76]: Since the flag key is the same as the key from the featureFlagForm can you loop through the flag keys and preform the check inside the loop?

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/create-or-update-feature-flags/create-or-update-feature-flags.component.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        featureFlagUpdateObj = {
+          name: this.featureFlagForm.get('name').value,
+          description: this.featureFlagForm.get('description').value,
+          flags
+        };
+      }
+      this.userData
+        .createOrUpdateFeatureFlags(JSON.stringify(featureFlagUpdateObj))
+        .subscribe( (response) => {
+          console.log(response);
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/create-or-update-feature-flags/create-or-update-feature-flags.component.ts:132]: Remove this console log.

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/delete-confirm-modal/delete-confirm-modal.component.ts [hunk 5]
```diff
@@ -9,12 +9,11 @@ import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
 export class DeleteConfirmModalComponent implements OnInit {
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/delete-confirm-modal/delete-confirm-modal.component.ts:9]: There will be a need for a deletion confirmation in many places. Can you make this reusable and move it to a shared location?

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/update-json/update-json.component.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  submitJSON() {
+    const inputJSON = this.jsonForm.get('featureflags').value;
+    const jsonList = inputJSON.split(',{');
+
+    jsonList.forEach((curr, i) => {
+      if (i >= 1) {
+        curr = '{' + curr;
+        console.log(curr);
+      }
```
Comment (Sbrenthughes) [inline src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/update-json/update-json.component.ts:42]: Why is this needed? Can it be done in a cleaner fashion ?

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts (hunk 2 above)
Comment (Courtneyp123) [reply src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts:60]: Would I be able to reduce them to just 
```
addFile() {
    const modalRef = this.modalService.open(UpdateJsonComponent);
    modalRef.result.then((newConfig) => {
        this.loadFeatureFlags();
    });
  }
```
So that I am able to call the loadFeatureFlags() after a result is given? (Just to refresh the page).

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/delete-confirm-modal/delete-confirm-modal.component.ts (hunk 5 above)
Comment (Courtneyp123) [reply src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/modal/delete-confirm-modal/delete-confirm-modal.component.ts:9]: Moved to shared.module.ts

File: src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts (hunk 2 above)
Comment (Courtneyp123) [reply src/app/admin_modules/admin_dashboard/dashboard/admin-dashboard/feature-flags/feature-flags.component.ts]: Pushed some changes for this.  Let me know if this is better.
